app/code/Magento/Customer/view/base/ui_component/customer_form.xml

```diff
@@ -269,6 +269,15 @@
                 <dataType>text</dataType>
                 <visible>true</visible>
             </settings>
+            <formElements>
+                <date>
+                    <settings>
+                        <options>
+                            <option name="maxDate" xsi:type="string">new Date()</option>
```

Comment:
Hi, @vishal-7037. Thanks for collaboration. XML declaration shouldn't contain any code that must be executed.   If we really need it, you can do it in the javascript class related to this component.